core/controllers/topics_and_skills_dashboard_test.py

```diff
@@ -239,14 +241,149 @@ def test_skill_creation_in_invalid_topic(self):
+            '"&quot;img.png&quot;" caption-with-value="&quot;&quot;" '
+            'alt-with-value="&quot;Image&quot;"></oppia-noninteractive-image>'
+        )
+        rubrics = [{
+            'difficulty': constants.SKILL_DIFFICULTIES[0],
+            'explanations': ['Explanation 1']
+        }, {
+            'difficulty': constants.SKILL_DIFFICULTIES[1],
+            'explanations': ['Explanation 2']
+        }, {
+            'difficulty': constants.SKILL_DIFFICULTIES[2],
+            'explanations': ['Explanation 3']
+        }]
+        post_data = {
+            'description': 'Skill Description',
+            'rubrics': rubrics,
+            'explanation_dict': state_domain.SubtitledHtml(
+                '1', explanation_html).to_dict(),
+            'thumbnail_filename': 'image.svg',
+            'filenames': [filename]
+        }
+
+        response_dict = self.post_json(
+            self.url, post_data,
+            csrf_token=csrf_token,
+            expected_status_int=400)
+
+        self.assertEqual(response_dict['error'], 'No image supplied')
+        self.logout()
+
+    def test_skill_creation_with_valid_images(self):
+        self.login(self.ADMIN_EMAIL)
+        csrf_token = self.get_new_csrf_token()
+        filename = 'img.png'
+        filename_2 = 'img_2.png'
+        explanation_html = (
+            '<oppia-noninteractive-image filepath-with-value='
+            '"&quot;img.png&quot;" caption-with-value="&quot;&quot;" '
+            'alt-with-value="&quot;Image&quot;"></oppia-noninteractive-image>'
+        )
+        explanation_html_2 = (
+            '<oppia-noninteractive-image filepath-with-value='
+            '"&quot;img_2.png&quot;" caption-with-value="&quot;&quot;" '
+            'alt-with-value="&quot;Image 2&quot;"></oppia-noninteractive-image>'
+        )
+        rubrics = [{
+            'difficulty': constants.SKILL_DIFFICULTIES[0],
+            'explanations': ['Explanation 1', explanation_html_2]
+        }, {
+            'difficulty': constants.SKILL_DIFFICULTIES[1],
+            'explanations': ['Explanation 2']
+        }, {
+            'difficulty': constants.SKILL_DIFFICULTIES[2],
+            'explanations': ['Explanation 3']
+        }]
+        post_data = {
+            'description': 'Skill Description',
+            'rubrics': rubrics,
+            'explanation_dict': state_domain.SubtitledHtml(
+                '1', explanation_html).to_dict(),
+            'thumbnail_filename': 'image.svg',
+            'filenames': [filename, filename_2]
+        }
+
+        with python_utils.open_file(
+            os.path.join(feconf.TESTS_DATA_DIR, 'img.png'),
+            mode='rb', encoding=None) as f:
+            raw_image = f.read()
+
+        self.post_json(
+            self.url, post_data,
+            csrf_token=csrf_token,
+            upload_files=(
+                (filename, filename, raw_image),
+                (filename_2, filename_2, raw_image),)
+        )
```

Comment:
Di we need to assert that this request has worked as expected?

Reply:
Done